Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -6555,6 +6555,13 @@ void WebPageProxy::didFinishLoadForFrame(FrameIdentifier frameID, FrameInfoData&
     RefPtr frame = WebFrameProxy::webFrame(frameID);
     MESSAGE_CHECK(m_process, frame);
 
+    // If a provisional load has since been started in another process, ignore this message.
+    if (m_preferences->siteIsolationEnabled()) {
+        RefPtr process = dynamicDowncast<WebProcessProxy>(AuxiliaryProcessProxy::fromConnection(connection));
+        if (frame->provisionalLoadProcess().coreProcessIdentifier() != process->coreProcessIdentifier())
```

Comment:
Can't you just do `if (frame->provisionalLoadProcess().connection() != &connection)`? Then you can skip the connection -> process lookup.

Reply:
hasConnection.  Great idea.